aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1378,17 +1377,19 @@ TORCH_IMPL_FUNC(mean_out)
     // which, in turn, does not produce as accurate results.
     bool is_half_type = (dtype == kHalf || dtype == kBFloat16);
     auto sum_out_dtype = is_half_type ? ScalarType::Float : dtype;
-    result_mut = is_half_type ? result_mut.to(sum_out_dtype) : result_mut;
+    auto result_temp = is_half_type ? result.to(sum_out_dtype) : result;
     // If dtype is FP16 or BF16, self (input tensor) will initially be cast to
     // FP32 in sum_out. This results in having to read that FP32 tensor again,
     // but maybe in the future, we could revise the implementation to not
     // materialize that intermediate FP32 tensor. That approach would probably
     // require some modifications in binary_kernel_reduce_vec(),
     // TensorIteratorBase::for_each(), and
     // TensorIteratorBase::serial_for_each(), apart from sum kernel for CPU.
-    at::sum_out(result_mut, self, opt_dim, keepdim, sum_out_dtype).div_(dim_prod);
-    // After sum & div, cast result_mut back to BF16 or FP16, if required.
-    result_mut = is_half_type ? result_mut.to(dtype) : result_mut;
+    at::sum_out(result_temp, self, opt_dim, keepdim, sum_out_dtype).div_(dim_prod);
+    // After sum & div, cast result_temp back to BF16 or FP16, if required.
+    if (is_half_type) {
+      result.copy_(result_temp.to(dtype));
```

Comment:
Do we need to call `result_temp.to(dtype)` ? result.copy_(result_temp) should works and save the overhead of `result_temp.to(dtype)`.

Reply:
Awesome， I just known copy_ do this convert. I have changed it.

Reply:
@CaoE - For accuracy reasons, wouldn't it be better if we have intermediate FP32 inputs we can use for division?